core/src/main/java/org/mule/DefaultMuleEvent.java

```diff
@@ -272,6 +299,16 @@ public DefaultMuleEvent(MuleEvent rewriteEvent, FlowConstruct flowConstruct)
             rewriteEvent.isSynchronous());
     }
 
+    /**
+     * Copy constructor used when ReplyToHandler instance needs switching out
+     * @param rewriteEvent
+     * @param replyToHandler
+     */
+    public DefaultMuleEvent(MuleEvent rewriteEvent, ReplyToHandler replyToHandler)
```

Comment:
For sure, but that is out of scope for this issue.  Let's consider later how important this is for 3.7 or not.